### ISSUE TITLE
Tagging shortcuts creates extra .lnk extension

### ISSUE THREAD

**Issue by chaicurioquest** [novoid/filetags#69]
I have created the shortcut in disk drive D:\ to the software installation folder in C: drive. I am not tagging the files inside the project folder (in disk drive C:) since it is unnecessary in the retrieval process.  I want to tag only workable/information files in C: drive through the shortcuts to classify/filter the projects. 

When I tried it I got the following issues

1. When I created the first tag1, it worked 
![Screenshot 2024-06-25 203429](https://github.com/novoid/filetags/assets/173520446/84c6e89b-124b-453b-9c89-25e3f853dae6)
2. But when I created the second tag2, it added an extra .lnk to the previous tag1 
![Screenshot 2024-06-25 203501](https://github.com/novoid/filetags/assets/173520446/1e4d5895-4951-4ec0-93d9-3862404ee425)
3. When I created tag3, it created an extra .lnk to  tag2 and removed .lnk from tag1. 
![Screenshot 2024-06-25 203600](https://github.com/novoid/filetags/assets/173520446/d6fd160b-98ba-4538-81e3-7a87e637d5e7)
But date2name script is working fine.
![Screenshot 2024-06-25 203621](https://github.com/novoid/filetags/assets/173520446/6d5279ee-ebba-4dc3-9008-f825bc61ffc0)
Is there any workaround/solution for this issue? Thanks.

**Comment by novoid** [comment]
@chaicurioquest Did you enable the Windows file browser option for showing file extensions or did you create the initial files without any extension at all?

**Comment by chaicurioquest** [comment]
@novoid, I did not enable Windows file browsing option. The initial file was created by software which I am using TeXtudio(Latex editor). I am only accessing the file using shortcuts pointing to that file. 
While executing `filetags --verbose --tags="tag1" chapter3.lnk`
DEBUG    2024-06-25 21:49:46,181 extracting list of files ...
DEBUG    2024-06-25 21:49:46,181 len(options.files) [1]
DEBUG    2024-06-25 21:49:46,181 WINDOWS: len(files) [1]
DEBUG    2024-06-25 21:49:46,181 WINDOWS: files CONVERTED [['chapter3.lnk']]
DEBUG    2024-06-25 21:49:46,181 1 filenames found: [chapter3.lnk]
DEBUG    2024-06-25 21:49:46,181 reported console width: 80 and height: 80   (80/80 is the fall-back)
DEBUG    2024-06-25 21:49:46,181 locate_and_parse_controlled_vocabulary: called with startfile: "chapter3.lnk"
DEBUG    2024-06-25 21:49:46,181 locate_and_parse_controlled_vocabulary: called in cwd: E:\Areas
DEBUG    2024-06-25 21:49:46,181 locate_file_in_cwd_and_parent_directories: called with startfile "chapter3.lnk" and filename ".filetags" ..
DEBUG    2024-06-25 21:49:46,181 locate_file_in_cwd_and_parent_directories: startfile [chapter3.lnk] found, using it as starting_dir [E:\Areas] ....
DEBUG    2024-06-25 21:49:46,181 locate_file_in_cwd_and_parent_directories: looking for ".filetags" in directory "E:\" .......
DEBUG    2024-06-25 21:49:46,181 locate_file_in_cwd_and_parent_directories: did NOT find ".filetags" in current directory or any parent directory
DEBUG    2024-06-25 21:49:46,181 locate_and_parse_controlled_vocabulary: locate_file_in_cwd_and_parent_directories did NOT find any filename
DEBUG    2024-06-25 21:49:46,181 locate_and_parse_controlled_vocabulary: this is Windows: also look out for lnk-files that link to .filetags files ...
DEBUG    2024-06-25 21:49:46,181 locate_file_in_cwd_and_parent_directories: called with startfile "chapter3.lnk" and filename ".filetags.lnk" ..
DEBUG    2024-06-25 21:49:46,181 locate_file_in_cwd_and_parent_directories: startfile [chapter3.lnk] found, using it as starting_dir [E:\Areas] ....
DEBUG    2024-06-25 21:49:46,181 locate_file_in_cwd_and_parent_directories: looking for ".filetags.lnk" in directory "E:\" .......
DEBUG    2024-06-25 21:49:46,181 locate_file_in_cwd_and_parent_directories: did NOT find ".filetags.lnk" in current directory or any parent directory
DEBUG    2024-06-25 21:49:46,181 locate_and_parse_controlled_vocabulary: this is Windows: .filetags (non-lnk) was found
DEBUG    2024-06-25 21:49:46,181 locate_and_parse_controlled_vocabulary: could not derive filename for controlled vocabulary
DEBUG    2024-06-25 21:49:46,181 non-interactive mode: extracting tags from argument ...
DEBUG    2024-06-25 21:49:46,181 tags found: [tag1]
DEBUG    2024-06-25 21:49:46,181 iterate over files ...
DEBUG    2024-06-25 21:49:46,181 determined maximum file name length with 12
DEBUG    2024-06-25 21:49:46,212 is_broken_link(chapter3.lnk) == D:\Latex-docs\PhD thesis\chapter3.pdf  which does exist -> non-broken link
DEBUG    2024-06-25 21:49:46,212 handle_file_and_optional_link("chapter3.lnk") …  ★★★★★★★★★★★★★★★★★★★★
DEBUG    2024-06-25 21:49:46,212 handle_file_and_optional_link: original directory = E:\Areas
DEBUG    2024-06-25 21:49:46,212 handle_file_and_optional_link: file is a non-broken link (and TAG_LINK_ORIGINALS_WHEN_TAGGING_LINKS is set)
DEBUG    2024-06-25 21:49:46,212 get_link_source_file(E:\Areas\chapter3.lnk) == D:\Latex-docs\PhD thesis\chapter3.pdf  which does exist -> non-broken link
DEBUG    2024-06-25 21:49:46,212 handle_file_and_optional_link: The file "E:\Areas\chapter3.lnk" is a link to "D:\Latex-docs\PhD thesis\chapter3.pdf" but they two do have different basenames. Therefore I ignore the original file.
DEBUG    2024-06-25 21:49:46,212 handle_file_and_optional_link: after handling potential link originals, I now handle the file we were talking about in the first place: E:\Areas\chapter3.lnk
DEBUG    2024-06-25 21:49:46,212 handle_file("E:\Areas\chapter3.lnk") ##########  … with working dir "E:\Areas"
DEBUG    2024-06-25 21:49:46,212 handle_file: set new_basename [chapter3.lnk] according to parameters (initialization)
DEBUG    2024-06-25 21:49:46,212 split_up_filename(chapter3) does NOT exist. Playing along and returning non-existent filename parts.
DEBUG    2024-06-25 21:49:46,212 adding_tag_to_filename(E:\Areas\chapter3.lnk, tag1): no tag found so far
DEBUG    2024-06-25 21:49:46,212 handle_file: set new_basename [E:\Areas\chapter3 -- tag1.lnk] when tagname != tag_in_unique_tags
  chapter3.lnk   ――renaming―→   chapter3 -- tag1.lnk
DEBUG    2024-06-25 21:49:46,212 handle_file("E:\Areas\chapter3.lnk") ##########  finished
DEBUG    2024-06-25 21:49:46,212 handle_file_and_optional_link: switching back to original directory = E:\Areas
DEBUG    2024-06-25 21:49:46,212 handle_file_and_optional_link("chapter3.lnk") FINISHED  ★★★★★★★★★★★★★★★★★★★★
DEBUG    2024-06-25 21:49:46,212 list_of_link_directories: ['E:\\Areas']
DEBUG    2024-06-25 21:49:46,212 successfully finished.
while executing tag2 `filetags --verbose --tags="tag2" chapter3.lnk` 
E:\Areas>filetags --verbose --tags="tag2" chapter3.lnk
DEBUG    2024-06-25 22:03:20,562 extracting list of files ...
DEBUG    2024-06-25 22:03:20,562 len(options.files) [1]
DEBUG    2024-06-25 22:03:20,562 WINDOWS: len(files) [0]
DEBUG    2024-06-25 22:03:20,562 WINDOWS: files CONVERTED [[]]
DEBUG    2024-06-25 22:03:20,562 0 filenames found: []
DEBUG    2024-06-25 22:03:20,562 reported console width: 80 and height: 80   (80/80 is the fall-back)
DEBUG    2024-06-25 22:03:20,562 locate_and_parse_controlled_vocabulary: called with startfile: "False"
DEBUG    2024-06-25 22:03:20,562 locate_and_parse_controlled_vocabulary: called in cwd: E:\Areas
DEBUG    2024-06-25 22:03:20,562 locate_file_in_cwd_and_parent_directories: called with startfile "E:\Areas" and filename ".filetags" ..
DEBUG    2024-06-25 22:03:20,562 locate_file_in_cwd_and_parent_directories: startfile [E:\Areas] is a directory, using it as starting_dir [E:\Areas] .....
DEBUG    2024-06-25 22:03:20,562 locate_file_in_cwd_and_parent_directories: looking for ".filetags" in directory "E:\" .......
DEBUG    2024-06-25 22:03:20,578 locate_file_in_cwd_and_parent_directories: did NOT find ".filetags" in current directory or any parent directory
DEBUG    2024-06-25 22:03:20,578 locate_and_parse_controlled_vocabulary: locate_file_in_cwd_and_parent_directories did NOT find any filename
DEBUG    2024-06-25 22:03:20,578 locate_and_parse_controlled_vocabulary: this is Windows: also look out for lnk-files that link to .filetags files ...
DEBUG    2024-06-25 22:03:20,578 locate_file_in_cwd_and_parent_directories: called with startfile "E:\Areas" and filename ".filetags.lnk" ..
DEBUG    2024-06-25 22:03:20,578 locate_file_in_cwd_and_parent_directories: startfile [E:\Areas] is a directory, using it as starting_dir [E:\Areas] .....
DEBUG    2024-06-25 22:03:20,578 locate_file_in_cwd_and_parent_directories: looking for ".filetags.lnk" in directory "E:\" .......
DEBUG    2024-06-25 22:03:20,578 locate_file_in_cwd_and_parent_directories: did NOT find ".filetags.lnk" in current directory or any parent directory
DEBUG    2024-06-25 22:03:20,578 locate_and_parse_controlled_vocabulary: this is Windows: .filetags (non-lnk) was found
DEBUG    2024-06-25 22:03:20,578 locate_and_parse_controlled_vocabulary: could not derive filename for controlled vocabulary
DEBUG    2024-06-25 22:03:20,578 non-interactive mode: extracting tags from argument ...
DEBUG    2024-06-25 22:03:20,578 tags found: [tag2]
DEBUG    2024-06-25 22:03:20,578 iterate over files ...
DEBUG    2024-06-25 22:03:20,578 determined maximum file name length with 0
DEBUG    2024-06-25 22:03:20,578 successfully finished.
**No update on tag2 in chapter3 -- tag1 during executing filetags in commandline**
But when I use GUI (through Windows explorer), tag2 is updated but tag1 is renamed with chapter3 -- tag1.lnk tag2

**Comment by novoid** [comment]
I'm sorry. I do have difficulties understanding what you mean.
What do you mean by "update on tag2 in chapter3", "tag2 is updated" or "tag1 is renamed"?

The first example output above doesn't seem to demonstrate any issue with false renaming. Therefore, I presume it doesn't help us with that issue as it seems.

The second example output above doesn't seem to demonstrate any renaming at all, especially not a wrong one.

I need one single example output with a wrong renaming process so that I can analyze your situation at hand. For that, I need (1) a clear information on the file name before, (2) the corresponding debug output and (3) the (wrong) file name afterwards. 

I don't get your last statement as well: do you mean that the result differs whether you are invoking filetags via the Windows Explorer integration and via the cmd.exe?

**Comment by chaicurioquest** [comment]
@novoid, 
When I execute, 
`E:\Areas>filetags --verbose --tags="tag1" *`, the output is 
<img width="576" alt="image" src="https://github.com/novoid/filetags/assets/173520446/e3982070-a4f0-4f60-8a45-e9e1c48f72b3">. No issues.

But second tagging `E:\Areas>filetags --verbose --tags="tag2" *`
<img width="577" alt="image" src="https://github.com/novoid/filetags/assets/173520446/0e34048f-c102-4c4e-a686-7504fcbc54ed">

Below is the verbose of the `E:\Areas>filetags --verbose --tags="tag2" *`
DEBUG    2024-06-26 22:26:06,182 extracting list of files ...
DEBUG    2024-06-26 22:26:06,182 len(options.files) [1]
DEBUG    2024-06-26 22:26:06,183 WINDOWS: len(files) [2]
DEBUG    2024-06-26 22:26:06,183 WINDOWS: files CONVERTED [['chapter 4 -- tag1.lnk', 'chapter3 -- tag1.lnk']]
DEBUG    2024-06-26 22:26:06,183 2 filenames found: [chapter 4 -- tag1.lnk], [chapter3 -- tag1.lnk]
DEBUG    2024-06-26 22:26:06,183 reported console width: 80 and height: 80   (80/80 is the fall-back)
DEBUG    2024-06-26 22:26:06,184 locate_and_parse_controlled_vocabulary: called with startfile: "chapter 4 -- tag1.lnk"
DEBUG    2024-06-26 22:26:06,184 locate_and_parse_controlled_vocabulary: called in cwd: E:\Areas
DEBUG    2024-06-26 22:26:06,184 locate_file_in_cwd_and_parent_directories: called with startfile "chapter 4 -- tag1.lnk" and filename ".filetags" ..
DEBUG    2024-06-26 22:26:06,186 locate_file_in_cwd_and_parent_directories: startfile [chapter 4 -- tag1.lnk] found, using it as starting_dir [E:\Areas] ....
DEBUG    2024-06-26 22:26:06,186 locate_file_in_cwd_and_parent_directories: looking for ".filetags" in directory "E:\" .......
DEBUG    2024-06-26 22:26:06,186 locate_file_in_cwd_and_parent_directories: did NOT find ".filetags" in current directory or any parent directory
DEBUG    2024-06-26 22:26:06,187 locate_and_parse_controlled_vocabulary: locate_file_in_cwd_and_parent_directories did NOT find any filename
DEBUG    2024-06-26 22:26:06,187 locate_and_parse_controlled_vocabulary: this is Windows: also look out for lnk-files that link to .filetags files ...
DEBUG    2024-06-26 22:26:06,187 locate_file_in_cwd_and_parent_directories: called with startfile "chapter 4 -- tag1.lnk" and filename ".filetags.lnk" ..
DEBUG    2024-06-26 22:26:06,188 locate_file_in_cwd_and_parent_directories: startfile [chapter 4 -- tag1.lnk] found, using it as starting_dir [E:\Areas] ....
DEBUG    2024-06-26 22:26:06,188 locate_file_in_cwd_and_parent_directories: looking for ".filetags.lnk" in directory "E:\" .......
DEBUG    2024-06-26 22:26:06,188 locate_file_in_cwd_and_parent_directories: found ".filetags.lnk" in directory "E:\" ........
DEBUG    2024-06-26 22:26:06,188 locate_and_parse_controlled_vocabulary: this is Windows: only a lnk of .filetags was found
DEBUG    2024-06-26 22:26:06,227 get_link_source_file(E:\.filetags.lnk) == C:\Users\Ram\.filetags  which does exist -> non-broken link
DEBUG    2024-06-26 22:26:06,230 locate_and_parse_controlled_vocabulary: this is Windows: set filename to source file for lnk .filetags
DEBUG    2024-06-26 22:26:06,235 get_link_source_file(E:\.filetags.lnk) == C:\Users\Ram\.filetags  which does exist -> non-broken link
DEBUG    2024-06-26 22:26:06,235 locate_and_parse_controlled_vocabulary: .filetags found: C:\Users\Ram\.filetags
DEBUG    2024-06-26 22:26:06,235 locate_and_parse_controlled_vocabulary: found controlled vocabulary
DEBUG    2024-06-26 22:26:06,235 locate_and_parse_controlled_vocabulary: reading controlled vocabulary in [C:\Users\Ram\.filetags]
DEBUG    2024-06-26 22:26:06,235 locate_and_parse_controlled_vocabulary: found unique tags: draft final
DEBUG    2024-06-26 22:26:06,237 locate_and_parse_controlled_vocabulary: found unique tags: confidential internal public
DEBUG    2024-06-26 22:26:06,237 locate_and_parse_controlled_vocabulary: found unique tags: scan screenshots
DEBUG    2024-06-26 22:26:06,237 locate_and_parse_controlled_vocabulary: controlled vocabulary has 21 tags
DEBUG    2024-06-26 22:26:06,237 locate_and_parse_controlled_vocabulary: controlled vocabulary has 3 groups of unique tags
DEBUG    2024-06-26 22:26:06,237 non-interactive mode: extracting tags from argument ...
DEBUG    2024-06-26 22:26:06,237 tags found: [tag2]
DEBUG    2024-06-26 22:26:06,237 iterate over files ...
DEBUG    2024-06-26 22:26:06,237 determined maximum file name length with 21
DEBUG    2024-06-26 22:26:06,242 is_broken_link(chapter 4 -- tag1.lnk) == D:\Latex-docs\PhD thesis\report.pdf  which does exist -> non-broken link
DEBUG    2024-06-26 22:26:06,242 handle_file_and_optional_link("chapter 4 -- tag1.lnk") …  ★★★★★★★★★★★★★★★★★★★★
DEBUG    2024-06-26 22:26:06,242 handle_file_and_optional_link: original directory = E:\Areas
DEBUG    2024-06-26 22:26:06,244 handle_file_and_optional_link: file is a non-broken link (and TAG_LINK_ORIGINALS_WHEN_TAGGING_LINKS is set)
DEBUG    2024-06-26 22:26:06,246 get_link_source_file(E:\Areas\chapter 4 -- tag1.lnk) == D:\Latex-docs\PhD thesis\report.pdf  which does exist -> non-broken link
DEBUG    2024-06-26 22:26:06,246 handle_file_and_optional_link: The file "E:\Areas\chapter 4 -- tag1.lnk" is a link to "D:\Latex-docs\PhD thesis\report.pdf" but they two do have different basenames. Therefore I ignore the original file.
DEBUG    2024-06-26 22:26:06,248 handle_file_and_optional_link: after handling potential link originals, I now handle the file we were talking about in the first place: E:\Areas\chapter 4 -- tag1.lnk
DEBUG    2024-06-26 22:26:06,248 handle_file("E:\Areas\chapter 4 -- tag1.lnk") ##########  … with working dir "E:\Areas"
DEBUG    2024-06-26 22:26:06,248 handle_file: set new_basename [chapter 4 -- tag1.lnk] according to parameters (initialization)
DEBUG    2024-06-26 22:26:06,249 split_up_filename(chapter 4 -- tag1) does NOT exist. Playing along and returning non-existent filename parts.
DEBUG    2024-06-26 22:26:06,249 split_up_filename(chapter 4 -- tag1) does NOT exist. Playing along and returning non-existent filename parts.
DEBUG    2024-06-26 22:26:06,249 adding_tag_to_filename(E:\Areas\chapter 4 -- tag1.lnk, tag2): add as additional tag to existing list of tags
DEBUG    2024-06-26 22:26:06,249 handle_file: set new_basename [E:\Areas\chapter 4 -- tag1.lnk tag2.lnk] when tagname != tag_in_unique_tags
  chapter 4 -- tag1.lnk   ――renaming―→   chapter 4 -- tag1.lnk tag2.lnk
DEBUG    2024-06-26 22:26:06,251 handle_file("E:\Areas\chapter 4 -- tag1.lnk") ##########  finished
DEBUG    2024-06-26 22:26:06,251 handle_file_and_optional_link: switching back to original directory = E:\Areas
DEBUG    2024-06-26 22:26:06,251 handle_file_and_optional_link("chapter 4 -- tag1.lnk") FINISHED  ★★★★★★★★★★★★★★★★★★★★
DEBUG    2024-06-26 22:26:06,251 list_of_link_directories: ['E:\\Areas']
DEBUG    2024-06-26 22:26:06,252 is_broken_link(chapter3 -- tag1.lnk) == D:\Latex-docs\PhD thesis\chapter3.pdf  which does exist -> non-broken link
DEBUG    2024-06-26 22:26:06,252 handle_file_and_optional_link("chapter3 -- tag1.lnk") …  ★★★★★★★★★★★★★★★★★★★★
DEBUG    2024-06-26 22:26:06,252 handle_file_and_optional_link: original directory = E:\Areas
DEBUG    2024-06-26 22:26:06,256 handle_file_and_optional_link: file is a non-broken link (and TAG_LINK_ORIGINALS_WHEN_TAGGING_LINKS is set)
DEBUG    2024-06-26 22:26:06,256 get_link_source_file(E:\Areas\chapter3 -- tag1.lnk) == D:\Latex-docs\PhD thesis\chapter3.pdf  which does exist -> non-broken link
DEBUG    2024-06-26 22:26:06,256 handle_file_and_optional_link: The file "E:\Areas\chapter3 -- tag1.lnk" is a link to "D:\Latex-docs\PhD thesis\chapter3.pdf" but they two do have different basenames. Therefore I ignore the original file.
DEBUG    2024-06-26 22:26:06,256 handle_file_and_optional_link: after handling potential link originals, I now handle the file we were talking about in the first place: E:\Areas\chapter3 -- tag1.lnk
DEBUG    2024-06-26 22:26:06,256 handle_file("E:\Areas\chapter3 -- tag1.lnk") ##########  … with working dir "E:\Areas"
DEBUG    2024-06-26 22:26:06,256 handle_file: set new_basename [chapter3 -- tag1.lnk] according to parameters (initialization)
DEBUG    2024-06-26 22:26:06,261 split_up_filename(chapter3 -- tag1) does NOT exist. Playing along and returning non-existent filename parts.
DEBUG    2024-06-26 22:26:06,261 split_up_filename(chapter3 -- tag1) does NOT exist. Playing along and returning non-existent filename parts.
DEBUG    2024-06-26 22:26:06,261 adding_tag_to_filename(E:\Areas\chapter3 -- tag1.lnk, tag2): add as additional tag to existing list of tags
DEBUG    2024-06-26 22:26:06,261 handle_file: set new_basename [E:\Areas\chapter3 -- tag1.lnk tag2.lnk] when tagname != tag_in_unique_tags
  chapter3 -- tag1.lnk    ――renaming―→   chapter3 -- tag1.lnk tag2.lnk
DEBUG    2024-06-26 22:26:06,263 handle_file("E:\Areas\chapter3 -- tag1.lnk") ##########  finished
DEBUG    2024-06-26 22:26:06,263 handle_file_and_optional_link: switching back to original directory = E:\Areas
DEBUG    2024-06-26 22:26:06,263 handle_file_and_optional_link("chapter3 -- tag1.lnk") FINISHED  ★★★★★★★★★★★★★★★★★★★★
DEBUG    2024-06-26 22:26:06,263 list_of_link_directories: ['E:\\Areas']
DEBUG    2024-06-26 22:26:06,263 successfully finished.

Please ignore the  comment in the previous post
"During the execution of tag2 filetags --verbose --tags="tag2" chapter3.lnk ,
There is no update on tag2 in chapter3 -- tag1 during executing filetags in commandline". I presume I wrongly executed filetags script in CLI. I should have executed `E:\Areas>filetags --verbose --tags="tag2" *`. Maybe, I'm not certain to execute filetags for individual files in CLI.  

I hope this post makes sense to you. Thank you for reply.

**Comment by novoid** [comment]
> E:\Areas>filetags --verbose --tags="tag2" *. Maybe, I'm not certain to execute filetags for individual files in CLI.

To tag a single file via the CLI, you use `E:\Areas>filetags --verbose --tags="tag2" E:\Areas\chapter3 -- tag1.lnk` for example. So you don't use the wildcard `*` but one or many file names.

**Comment by novoid** [comment]
Good news, I can reproduce the bug in Linux as well:

```
2024-06-28-debug-filetags-issue-69 % touch "chapter 4 -- tag1.lnk"
2024-06-28-debug-filetags-issue-69 % l
drwxr-xr-x vk users 4.0 KB Fri Jun 28 10:48:11 2024  .
drwxrwxr-x vk users  24 KB Fri Jun 28 10:48:07 2024  ..
.rw-r--r-- vk users   0 B  Fri Jun 28 10:48:11 2024  'chapter 4 -- tag1.lnk'
2024-06-28-debug-filetags-issue-69 % t -t "tag2" chapter\ 4\ --\ tag1.lnk 
  chapter 4 -- tag1.lnk   ――renaming―→   chapter 4 -- tag1.lnk tag2.lnk
2024-06-28-debug-filetags-issue-69 % l
drwxr-xr-x vk users 4.0 KB Fri Jun 28 10:48:26 2024  .
drwxrwxr-x vk users  24 KB Fri Jun 28 10:48:07 2024  ..
.rw-r--r-- vk users   0 B  Fri Jun 28 10:48:11 2024  'chapter 4 -- tag1.lnk tag2.lnk'
2024-06-28-debug-filetags-issue-69 % touch test.lnk
2024-06-28-debug-filetags-issue-69 % t test.lnk 
                 
Please enter tags, separated by " "; abort with Ctrl-C; complete 29 tags with TAB
                     
         .--------, 
        | o   ?   | 
         `--------' 
                     

  Previously used tags in this directory:
    tag1.lnk (1) ⋅ tag2 (2)

Tags: foo

  test.lnk   ――renaming―→   test -- foo.lnk
2024-06-28-debug-filetags-issue-69 % l
drwxr-xr-x vk users 4.0 KB Fri Jun 28 10:48:47 2024  .
drwxrwxr-x vk users  24 KB Fri Jun 28 10:48:07 2024  ..
.rw-r--r-- vk users   0 B  Fri Jun 28 10:48:11 2024  'chapter 4 -- tag1.lnk tag2.lnk'
.rw-r--r-- vk users   0 B  Fri Jun 28 10:48:43 2024  'test -- foo.lnk'
2024-06-28-debug-filetags-issue-69 % 
```

I'll debug this further and provide a fix.

**Comment by novoid** [comment]
@chaicurioquest Please do test the most recent version from today. I'm pretty sure I found the bug and fixed it. Report back if the issue still persists.

Have a nice weekend!

**Comment by chaicurioquest** [comment]
@novoid Thank you. I will try with an updated code. 👍

**Comment by chaicurioquest** [comment]
It works. Thank you for your time and work.

**Comment by nbehrnd** [comment]
@chaicurioquest Independent that the problem you report for Windows seems to be resolved, the very first screen photo looks like the files you attempted to tag were neither plain illustrations, nor .tex source code, but links.  Though links/bookmarks to what, individual chapters of an (eventually compiled to a single file) pdf (as the icons suggest)?  Are they of temporary nature (like .log and .aux) to build the large pdf?

So far I presumed tagging pdfs could perhaps be sensible as parts of a compiled/staple thesis, i.e. where parts of «the book» don't have to be (re)written if there are successful publications during the PhD sufficiently sum up (individual) background, results and discussion. Then, there would be only a small introduction in the corresponding .tex file ahead of their inclusion (for instance by [pdfpages](https://ctan.org/pkg/pdfpages)) and files around writing the thesis would be organized like this (constructed) tree diagram:

```
.
├── main.tex
├── general_introduction.tex
├── terrestial_planets.tex
├── exoplanets.tex
├── experiments.tex
├── conclusion.tex
├── literature.bib
├── illustrations
│   ├── moons_of_earth.png
│   ├── moons_of_iupyter.png

├── program_source_codes
│   ├── regression_analysis.py
│   ├── principal_component_analysis.f90

└── papers_published
    ├── special_theory_paper.pdf
    ├── generalized_theory_paper.pdf
```

**Comment by chaicurioquest** [comment]
@nbehrnd, I have used a thesis report on latex as an example for shortcut tagging. I used the shortcut .lnk file(in E:\ drive) to execute the main latex file, here it's only **report.tex** (in D:\ drive). The above examples (chapter 3 was taken just for demonstration purposes only) Since I don't want to tag all latex-related files generated.  I am not using any CTAN package. I include all chapters and sections using \include (as mentioned in the screenshot below),
![Screenshot 2024-07-03 110027](https://github.com/novoid/filetags/assets/173520446/e847816d-0944-4118-bcf8-ef9eb46c03b5)
![Screenshot 2024-07-03 105947](https://github.com/novoid/filetags/assets/173520446/5529d4ea-205d-4a1a-b19c-7449f6120837)
I have the following individual .tex files. The main .tex file(report.tex) will have the shortcut on other drive E:\. Yeah, the publication is part of PhD work; I have added it in the appendix.
![Screenshot 2024-07-03 105107](https://github.com/novoid/filetags/assets/173520446/2676abb3-6368-4001-8438-f3ef1dd086d8)
